src/components/structures/RoomSearchView.tsx

```diff
@@ -283,6 +303,17 @@ export const RoomSearchView = forwardRef<ScrollPanel, Props>(
                 ourEventsIndexes.push(result.context.getOurEventIndex());
             }
 
+            let permalinkCreator = _permalinkCreator;
+            if (roomId !== permalinkCreator.roomId) {
```

Comment:
can we just create a new `permalinkcreator` even if it's a single-room search, rather than having to pass one in? seems like it would simplify a lot of logic.